Assets/Packages/com.studio23.ss2.splashscreensystem/Runtime/Core/SplashScreenBehaviour.cs

```diff
@@ -0,0 +1,59 @@
+using Cysharp.Threading.Tasks;
+using Studio23.SS2.SplashScreenSystem.Data;
+using Studio23.SS2.SplashScreenSystem.UI;
+using System;
+using UnityEngine;
+
+namespace Studio23.SS2.SplashScreenSystem.Core
+{
+    public class SplashScreenBehaviour : MonoBehaviour
+    {
+        public SplashScreenData[] splashScreens;
+        public event Action<bool> OnFinish;
+        public int currentIndex = 0;
+
+        private void Start()
+        {
+            ShowSplashScreen();
+        }
+
+        private async void ShowSplashScreen()
+        {
+            if (currentIndex < splashScreens.Length)
+            {
+                SplashScreenData currentSplash = splashScreens[currentIndex];
+                if (currentSplash.data != null)
+                    SplashScreenUIManager.Instance.DisplayData(currentSplash.data);
+                CrossFadeScreen(currentSplash.fadeduration);
+                if (currentSplash.data.name != "EULA")
+                {
+                    await UniTask.Delay(TimeSpan.FromSeconds(currentSplash.duration));
+                    HideSplashScreen();
+                }
+            }
+            else
+            {
+                OnFinish?.Invoke(true);
+                currentIndex = 0;
+            }
+        }
+
+        public void HideSplashScreen()
+        {
+            currentIndex++;
+            ShowSplashScreen();
+        }
```

Comment:
Instead of hiding the loop by splitting into two functions and manually handlign the index, just use a foreach loop.